src/backend/src/yaksuri_progress.c

```diff
@@ -1065,6 +1065,42 @@ static int set_multichunk_subreq(const void *inbuf, void *outbuf, uintptr_t coun
     return rc;
 }
 
+/* Fast-path pack subroutine */
+static int singlechunk_pack(yaksuri_gpudriver_id_e id, int device, const void *inbuf, void *outbuf,
+                            uintptr_t count, yaksi_type_s * type, yaksi_info_s * info,
+                            yaksa_op_t op, yaksuri_subreq_s * subreq)
```

Comment:
The name need to reflect this is for same-device transfer (is it?).

Reply:
Not really. The current code only uses it for same-device fast path. But we will also use it to optimize the IPC and P2P-enabled cases in the upcoming optimization PR.

Reply:
okay

Reply:
An update for the naming of `singlechunk_pack`: after second read on this code, this routine is used only for `YAKSA_OP__REPLACE` in the current code. It is still unclear to me whether the same fast path can be used for other ops (e.g., MAX, MIN). But I guess it is OK to keep the name for now. Once we make optimization for non-REPLACE ops, we can adjust at that time.